core/src/main/java/org/elasticsearch/index/seqno/GlobalCheckpointService.java

```diff
@@ -67,16 +69,20 @@
      */
     final private ObjectLongMap<String> trackingLocalCheckpoint;
 
-    private long globalCheckpoint = SequenceNumbersService.UNASSIGNED_SEQ_NO;
+    private long globalCheckpoint;
+
+    public GlobalCheckpointService(final ShardId shardId, final IndexSettings indexSettings) {
```

Comment:
I pushed a9d5f5e079c21cb541c39a000e9bfe2a85b2f769 and 6e8f11ae3da78c4b82cd267c1daba3951b462eee.